core/services/workflows/metering/metering.go

```diff
@@ -166,67 +207,93 @@ func (r *Report) MedianSpend() map[SpendUnit]SpendValue {
 	return medians
 }
 
-// ReserveStep earmarks the maximum spend for a given capability invocation in the engine.
+// ReserveByLimits earmarks an amount of local universal credit balance and then returns that amount
+// The amount reserved is determined by the upper limit of resource credits that can be used
 // We expect to only set this value once - an error is returned if a step would be overwritten
-func (r *Report) ReserveStep(ref ReportStepRef, capInfo capabilities.CapabilityInfo) error {
+func (r *Report) ReserveByLimits(ref string, capInfo capabilities.CapabilityInfo, limits []SpendTuple) (int64, error) {
```

Comment:
This function seems to only be used for the initial compute reserve. Would it be possible to condense this into the initialize step, leaving only a single reserve call? Having a single reserve would simplify the interface.